### ISSUE TITLE
Do losslink nodes apply correctly the lossfactor?

### ISSUE THREAD

**Issue by BaptisteFrancois** [pywr/pywr#1121]
Hi,
The documentation of the `losslink` node defines the loss factor as:
```The proportion of flow that is lost through this node. Must be greater than or equal to zero. If zero then no-losses are calculated. The percentage is calculated as a percentage of gross flow.```

I understand that the fraction that is lost is a percentage of the flow entering the node (i.e., the gross flow). For instance, if 1 MGD enters the node, should the loss factor be 0.2, the node should outflow 0.8 MGD (i.e., we lost 20% of the input). However, this is not the case. It actually outflows 0.833333.

The thing is that the outflow of the `losslink` is currently calculated as `gross_flow / (1+loss_factor)`. (given my example, it means `gross_flow/1.2` or, `gross_flow*0.8333`, which to me does not align with the above definition of the loss factor. In the above example, although the loss factor is 0.2, we actually loose 16.7% of the gross flow, not 20%.

Given the explanation above (i.e., the loss factor being `the proportion of flow that is lost through this node` and `The percentage is calculated as a percentage of gross flow`, I would rather think that the outflow of the `losslink` to be calculated as: `gross_flow * (1-loss_factor)`.

Whether you agree with me or not, the equation about how the loss factor is applied to obtain the net flow would help clarify what the node does.

Below is a json code to reproduce the example I mentioned above:

```
{
    "metadata": {
      "title": "demo_losslink_bug",
      "description": "A simple model to demonstrate the issue with losslink",
      "author": "Baptiste Francois - bfrancois@umass.edu",
      "minimum_version": "1.21.0"
    },
  
    "timestepper": {
      "start": "2024-04-05",
      "end": "2024-04-15",
      "timestep": 1
    },

    "nodes": [

        {
            "name": "input_1",
            "type": "catchment",
            "flow": 1
        },
        {
            "name": "faulty_losslink",
            "type": "losslink",
            "loss_factor": 0.2
        },
        {
            "name": "demand",
            "type": "output",
            "max_flow": 1,
            "cost": -1
        }
    ],
    "edges": [
        ["input_1", "faulty_losslink"],
        ["faulty_losslink", "demand"]
    ],
    "recorders": {
        "input": {
            "type": "numpyarraynoderecorder",
            "node": "input_1"
        },
        "net_input": {
            "type": "numpyarraynoderecorder",
            "node": "faulty_losslink"
        },
        "output": {
            "type": "numpyarraynoderecorder",
            "node": "demand"
        }
    }
}```

**Comment by jetuk** [comment]
I agree this is poorly documented. It was implemented to apply the factor relative to the net/output flow. I have a WIP branch improving the documentation and allowing the user to defined whether the factor is gross or net. However, I never finished it. Given your issue I will tidy it up and hopefully we can agree on an implementation.

**Comment by jetuk** [comment]
@BaptisteFrancois are you able to have a look at or test the changes in #1124 to see if it clarifies things and/or adds the features you need?

**Comment by BaptisteFrancois** [comment]
@jetuk looking at the code, it seems to add the feature I needed. However, I failed to install the `loss-link-loss-factor-type` branch from the src so that could not test it.

**Comment by jetuk** [comment]
Fixed in #1124
